addons/cert.py

```diff
@@ -235,6 +235,42 @@ def msc30(data):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
 
+# STR11-C
+# Do not specify the bound of a character array initialized with a string literal
+def str11(data):
+    for token in data.tokenlist:
+        if not token.isString:
+            continue
+
+        strlen = token.strlen
+        parent = token.astParent
+
+        if parent is None:
+            continue
+        parentOp1 = parent.astOperand1
+        if parentOp1 is None or parentOp1.str!='[':
+            continue
+
+        if not parent.isAssignmentOp:
+            continue
+            
+        varToken = parentOp1.astOperand1
+        if varToken is None or not varToken.isName:
+            continue
+        if varToken.variable is None:
+            continue
+        startToken = varToken.variable.typeStartToken   
+        endToken = varToken.variable.typeEndToken # check if it's the core variable declaration
+        if startToken != endToken:
```

Comment:
Here you want to check if it's a variable declaration or not and then I think this code is better:

    if varToken.variable.nameToken == varToken:
        # This is the declaration for varToken.variable
        continue

Reply:
If I change the code as you mentioned the test will fail!? Do I miss something here?

Reply:
My fault.. invert the condition:

        if varToken != varToken.variable.nameToken:
            continue